### ISSUE TITLE
Decorators: conflict between no-multi-spaces and key-spacing rules

### ISSUE THREAD

**Issue by nuno-m** [typescript-eslint/typescript-eslint#2981]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
I'm not entirely sure of where to place this issue or how to describe correctly the title for this issue.
There's already a similar issue on ESLint repo (https://github.com/eslint/eslint/issues/12788), and someone over there recommended reaching out someone here.
The below set of rules are working as intended everywhere on my code, except on Typescript `@Decorators` definitions.
If I format the code as I wish, the parser complains about the `no-multi-spaces` rule, but if I remove the spaces to fix it, it complains about the `key-spacing` rule.
I know these rules do not belong to this repo, but since they are working as expected, except on decorators, I thought maybe this would be the place to ask.
I temporarily solved my problem by ignoring the `no-multi-spaces` on the decorator, but I was wondering if it's possible to handle this Typescript case without disabling any rules.

```JSON
{
  "rules": {
    "no-multi-spaces": ["error"],
    "key-spacing": [
      "error",
      {
        "beforeColon": false,
        "afterColon": true,
        "mode": "minimum",
        "align": "value"
      }
    ]
  }
}
```

```TS
@NgModule({
  declarations: [AppComponent],
  imports:      [BrowserModule],
  providers:    [],
  bootstrap:    [AppComponent]
})
export class AppModule {
}
```

**Expected Result**
No linting error (no conflict between `no-multi-spaces` and `key-spacing` rules)

**Actual Result**
Error on `no-multi-spaces` rule.
If the error is fixed for this rule, then there's an error for `key-spacing` rule.

**Versions**
| package                            | version   |
| ---------------------------------- | --------- |
| `@typescript-eslint/eslint-plugin` | `4.14.0`  |
| `@typescript-eslint/parser`        | `4.14.0`  |
| `TypeScript`                       | `4.1.3`   |
| `ESLint`                           | `7.18.0`  |
| `node`                             | `14.15.4` |

**Comment by bradzacher** [comment]
The only solution is to fork this rule.

**Comment by nuno-mrks** [comment]
OK, thanks for the feedback.
So, the current approach for now is to disable `no-multi-spaces` on the decorators, if I want to keep my code formatted that way, right?

**Comment by looch** [comment]
> OK, thanks for the feedback.好的，谢谢你的反馈。 So, the current approach for now is to disable `no-multi-spaces` on the decorators, if I want to keep my code formatted that way, right? 所以，目前的方法是禁用装饰器上的 `no-multi-spaces` ，如果我想保持代码的格式，对吗？

😱Nobody fixed this bug?

**Comment by bradzacher** [comment]
Hi there 👋!

As you may have heard [ESLint core is deprecating its formatting rules](https://eslint.org/blog/2023/10/deprecating-formatting-rules/). We are likewise moving in the same direction and deprecating our formatting rules.

Deprecated rules will be removed in the next major version as their new home is the [eslint-stylistic](https://github.com/eslint-stylistic/eslint-stylistic) project.

If this issue is important to you please consider filling an issue over at eslint-stylistic which will be the new home for stylistic/formatting rules.

Thanks!

**Comment by cyonder** [comment]
I'm also having the same problem @bradzacher 

I put together a failing code for `eslint-stylistic` but for some reason when I run `pnpm run play:lint` I don't get eslint warning, even for a semicolon. Although, I get the warnings in `eslint.config.js`. Anyway, if you could run the eslint, you should see the warning. Unfortunately, I didn't wanna open an issue on `eslint-stylistic` since I couldn't get the demo working for some reason, so sending it here.

https://stackblitz.com/edit/eslint-stylistic-eslint-stylistic-g2n29d?file=playground%2Feslint.config.js,playground%2Findex.ts
